plugins/out_pgsql/pgsql.c

```diff
@@ -0,0 +1,297 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2020 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_output.h>
+#include <fluent-bit/flb_pack.h>
+
+#include "pgsql.h"
+
+void pgsql_conf_destroy(struct flb_pgsql_config *ctx)
+{
+    PGresult *res = NULL;
+
+    if(PQstatus(ctx->conn) == CONNECTION_OK) {
+        while(PQconsumeInput(ctx->conn) == 0) {
+            res = PQgetResult(ctx->conn);
+            if(PQresultStatus(res) != PGRES_COMMAND_OK) {
+                flb_warn("[out_pgsql] %s", PQerrorMessage(ctx->conn));
+            }
+            PQclear(res);
+        }
+    }
+
+    flb_free(ctx->db_hostname);
+
+    if(ctx->db_table != NULL) {
+        flb_sds_destroy(ctx->db_table);
+    }
+
+    if(ctx->timestamp_key != NULL) {
+        flb_sds_destroy(ctx->timestamp_key);
+    }
+
+    PQfinish(ctx->conn);
+
+    free(ctx);
```

Comment:
@edsiper @sxd should this be `free` or `flb_free`?

Reply:
good question! I think it should be flb_free() but since I'm used to use free() that may be the reason why I wrote this... let's use flb_free() =)